cips/cip-6-second-blocks.md

```diff
@@ -0,0 +1,60 @@
+| cip | TBD |
+| - | - |
+| title | Reduce block time to six seconds |
+| description | Proposal to reduce block time on celestia-app to 6 seconds, from 12. |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)) |
+| discussions-to | <https://forum.celestia.org/t/cip-decrease-block-time-to-6-seconds/1836> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-09 |
+
+## Abstract
+
+This CIP proposes to reduce the block time on celestia-app to 6 seconds, from 12. This will increase the throughput by double and and reduce the time it takes for transactions to be finalized by half.
+
+## Motivation
+
+The motivation for this CIP stems from a discussion in Core Devs Call 17, where it was proposed to reduce the block time to 6 seconds from 12 seconds.
+
+## Specification
+
+1. The block time in celestia-app MUST be reduced from 12 seconds to 6 seconds.
+
+2. All implementations of celestia-app SHALL adjust their block production mechanisms to conform to this new 6-second block time.
+
+3. The change in block time MUST be implemented at a specific block height, which SHALL be determined and agreed upon by the Celestia community.
+
+4. Celestia consensus nodes SHOULD update their software to accommodate this change prior to the agreed-upon block height.
+
+5. The reduction in block time SHALL NOT affect the overall security or integrity of the Celestia network.
+
+6. Validators MUST ensure that their systems are capable of producing and validating blocks at this increased rate.
+
+7. Client applications interacting with the Celestia network SHOULD be updated to account for the faster block time, particularly in areas related to transaction confirmation and block finality.
+
+8. The implementation of this change SHOULD include appropriate testing and monitoring to ensure network stability during and after the transition.
+
+9. Documentation and APIs related to block time and block production MUST be updated to reflect this change.
+
+10. The network SHOULD provide a grace period for node operators and other participants to adapt to the new block time, but all participants MUST be compliant by the agreed-upon implementation block height.
```

Comment:
This item is more relevant for https://github.com/celestiaorg/CIPs/blob/main/cips/cip-10.md and the upgrade mechanism rather than this CIP. Alternatively it could even be placed in the meta-CIP for ginger.

```suggestion
```

Reply:
I'll revisit this later today. to clarify, you mean including CIP-10 as part of the Meta CIP for Ginger?

Reply:
To clarify I mean that CIP-10 is the right place for discussion on the "grace period for node operators". Since that CIP already merged and got included, we can just drop this line from this CIP.

I don't think CIP-10 needs to be part of the meta CIP.

FWIW there is actually a grace period of one week after 5/6 validators have voted for a version but that was added after CIP-10 was created.

Reply:
made an issue to track this https://github.com/celestiaorg/CIPs/issues/218